parley/src/layout/mod.rs

```diff
@@ -280,6 +280,23 @@ pub struct Line<'a, B: Brush> {
     data: &'a LineData,
 }
 
+#[derive(Clone, Debug)]
+pub(crate) enum LayoutLineHeight {
+    MetricsRelative(f32),
+    Absolute(f32),
+}
+
+impl LayoutLineHeight {
+    pub(crate) fn resolve(&self, run: &RunData) -> f32 {
+        match self {
+            Self::MetricsRelative(value) => {
+                (run.metrics.ascent + run.metrics.descent + run.metrics.leading) * value
```

Comment:
I only just found out what `leading` is - it seems to be made for exactly this use case!

I wonder if `value` should actually be a multiplier on `leading`. It's probably too cute. Landing this as-is would be fine

Reply:
For what it's worth, this summing is how the metrics based line height was calculated in [the original Chad code](https://github.com/linebender/parley/blob/3356fd4cf0bda07b95dcd697ee8327e5cd95e4e0/src/layout.rs#L550-L553).

Reply:
`leading` is just the font-specified extra gap between lines, beyond the ascender and descender height. It's quite small compared to other metrics, and could vary a lot between fonts.

Say one font has an ascent+descent of 700 and a leading of 300, and another has an ascent+descent of 900 and a leading of 100. Multiplying just the leading would result in `LineHeight::MetricsRelative` being three times as sensitive on the former font than the latter, despite both having the same line height. Theoretically it could also be zero or negative (the OpenType property is a signed value) which would result in more weirdness.

Reply:
This was largely idle musing - I agree it shouldn't be actioned.